media-libs/dav1d/dav1d-9999.ebuild

```diff
@@ -0,0 +1,52 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+SCM=""
+if [[ "${PV}" == "9999" ]]; then
+	SCM="git-r3"
+	EGIT_REPO_URI="https://code.videolan.org/videolan/dav1d"
+else
+	KEYWORDS="~amd64"
+	SRC_URI="https://code.videolan.org/videolan/dav1d/-/archive/${PV}/${P}.tar.bz2"
+fi
+
+inherit ${SCM} meson ninja-utils multilib-minimal
+
+DESCRIPTION="dav1d is an AV1 Decoder :)"
+HOMEPAGE="https://code.videolan.org/videolan/dav1d"
+
+LICENSE="BSD-2"
+SLOT="0"
+IUSE="+8bit +10bit +asm"
+
+ASM_DEPEND=">=dev-lang/nasm-2.13"
+RDEPEND=""
+DEPEND="${RDEPEND}
+	asm? (
+		abi_x86_32? ( ${ASM_DEPEND} )
+		abi_x86_64? ( ${ASM_DEPEND} )
+	)"
+
+DOCS=( README.md doc/PATENTS )
+
+multilib_src_configure() {
+	local -a bits=()
+	use 8bit  && bits+=( 8 )
+	use 10bit && bits+=( 10 )
+
+	local emesonargs=(
+		-D bitdepths=$(IFS=,; echo "${bits[*]}")
+		-D build_asm=$(usex asm true false)
```

Comment:
How does this interact with conditional deps above?

Reply:
My current understanding is that `build_asm=true` enables building assembly in general. What this means depends on the platform. On x86, it requires nasm. On all other platforms, it's currently a no-op. I do not know if nasm will be required if/when they add assembly for other platforms, since nasm is x86-specific. @ePirat please comment if what I said is correct?

Reply:
Ok, let me be more specific of what I was thinking: will this explode on `abi_x86_x32`? ;-)

Reply:
> My current understanding is that build_asm=true enables building assembly in general. 

@haasn: This is true.

Reply:
Oh, that's a good question. According to the meson.build script, the nasm dependency is added if `host_machine.cpu_family().startswith('x86'))`. Since I assume this is still the case for abi_x86_x32, we need to add the nasm dependency to that architecture as well. Do you want to add it or shall I submit a PR?

Reply:
I honestly doubt the assembly will work on x32 if it's not officially supported. You may need to enable asm conditionally to MULTILIB_ABI_FLAG var. I'll test it later today.